submissions/the-darkesttt/js-post-oop/index.js

```diff
@@ -0,0 +1,64 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/the-darkesttt/a-tiny-JS-world
+   Web app: https://the-darkesttt.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+
+class Creature {
+
+   constructor( name, gender, saying, species) {
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.species = species;
+   }
+}
+
+class Animal extends Creature {
+   constructor(name, gender, saying, paws, species) {
+      super(name, gender, saying, species);
+      this.paws = paws;
+   }
+
+   showValues() {
+      return `${this.species}; ${this.name}; ${this.gender}; ${this.saying}; ${this.paws}`
```

Comment:
Make this DRY suing `Array#map`.
Also, class should handle its own properties. Encapsulation from OOP and Open/Closed from SOLID.

Reply:
Explain, why ignored.
Imagine, you are on an important commercial project.